### PR TITLE
Combine keys and datasets in Manifest instantiation

### PR DESCRIPTION
**Pull request by cortadocodes**
## Summary
Simplify instantiation of manifests by specifying their datasets as a dictionary instead of a list with a separate keys parameter.

## Contents ([#355](https://github.com/octue/octue-sdk-python/pull/355))
**IMPORTANT:** There are 3 breaking changes.

### Enhancements
- **BREAKING CHANGE:** Subsume `keys` parameter into `datasets` parameter for `Manifest` instantiation by making `datasets` a dictionary
- **BREAKING CHANGE:** Serialise a manifest's datasets as a dictionary
- **BREAKING CHANGE:** Use latest schema for datasets in manifests

### Dependencies
- Use `twined==0.2.0`

### Documentation
- Use correct GitHub workflow badge in `README`

### CODE REVIEW

File: octue/resources/manifest.py [hunk 1]
```diff
@@ -13,34 +13,48 @@
...
+    :param str|None path: the path the manifest exists at (defaults to the current working directory)
+    :param dict|None datasets: a mapping of dataset names to `Dataset` instances, serialised datasets, or paths to datasets
+    :return None:
     """
 
     _ATTRIBUTES_TO_HASH = ("datasets",)
     _SERIALISE_FIELDS = "datasets", "keys", "id", "name", "path"
 
-    def __init__(self, id=None, path=None, datasets=None, keys=None, **kwargs):
+    def __init__(self, id=None, path=None, datasets=None, **kwargs):
```
Comment (thclark) [inline octue/resources/manifest.py:26]: I don't feel right about having this here in the init method. We'll talk more about why that is in the coming months (work with the IEA will draw out the need to translate and provide transformers between schema versions).

Can you actually pull it right out into a function, perhaps in a 'migrations" module?

That makes it very clear what it's for - keeping the piece of code we have to deprecate out of the main program logic but also providing me a reference point that I can think about establishing a more formalised data migration process on top of.

**Review by thclark — CHANGES_REQUESTED** [review]
Only one requested change, basically so I can start crystallising what a more formal data migration process should look like

File: octue/resources/manifest.py (hunk 1 above)
Comment (cortadocodes) [reply octue/resources/manifest.py:26]: Yep will do. Will it be removed from the `migrations` module in the next breaking change release?

Comment (cortadocodes) [reply octue/resources/manifest.py:26]: Also did you want the same thing done in e.g. `twined` next time?

**Review by thclark — APPROVED** [review]

**Pull request merged**